### PR TITLE
[Fix] - Fix Create Product API

### PR DESCRIPTION
**Pull request by KezTimple**
## Ticket/s

- [Failed Ticket](https://trello.com/c/BQvSsb8c/32-api-create-product)

## Changes

- Fixed product ID (no spaces)

## Screenshot

- Product ID with space
image

## How to test?

- Open Postman and copy cURL to Create Product
```
curl --location 'http://localhost:3001/products/create' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6IkFscGhhIiwic3ViIjoiNjRkMGExZDg2MzhhNzA4YjgzZTRlZDNjIiwidXNlcnR5cGUiOiJtZXJjaGFudCIsImlhdCI6MTY5MTQ2MjQ1MSwiZXhwIjoxNjkxNTQ4ODUxfQ.Iw4WVjnGAUuhRlJnkQey8Wqr83FBJU97-JUXdVC7u-k' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "product 01234",
    "productImg": {
        "ImgURL": "https://example.com/image.jpg",
        "ImgAttch": "base64encodedimage"
    },
    "productName": "Product Name",
    "productInfo": "Product Information",
    "productPrice": 199.99,
    "productInventory": 32,
    "productCategory": [
        "sample",
        "sample1",
        "sample2"
    ]
}'
```

### CODE REVIEW

**Review by DanDalanDone69 — APPROVED** [review]
LGTM!

**Comment by ianchinjen** [comment]
Please remove authorization from the logic as it is on a different ticket.

**Review by sandrex-SGV — CHANGES_REQUESTED** [review]
Remove product/merchant guard to avoid conflict with authorization ticket as it is being developed by sir @ianchinjen 

[API User Authorization](https://trello.com/c/IjnkaW52/3-api-user-user-authorization)

also remove userType in payload

File: apps/api/src/auth/auth.service.ts [hunk 1]
```diff
@@ -33,7 +33,11 @@ export class AuthService {
       throw new UnauthorizedException();
     }
 
-    const payload = { username: user.username, sub: user._id };
+    const payload = {
+      username: user.username,
+      sub: user._id,
+      usertype: user.userType,
```
Comment (sandrex-SGV) [inline apps/api/src/auth/auth.service.ts]:
```suggestion
      
```
 remove `userType` in the payload

File: apps/api/src/products/products.controller.ts [hunk 2]
```diff
@@ -6,11 +6,14 @@ import {
   HttpCode,
   HttpStatus,
   Get,
+  UseGuards,
```
Comment (sandrex-SGV) [inline apps/api/src/products/products.controller.ts]:
```suggestion
  
```
remove `UseGuards`

File: apps/api/src/products/products.controller.ts [hunk 3]
```diff
@@ -6,11 +6,14 @@ import {
...
 } from '@nestjs/common';
 import { ProductsService } from './products.service';
 import { CreateProductDto } from './dto/create-product.dto';
 import { Product } from './schemas/products.schema';
+import { MerchantGuard } from './product.guard';
 
+@UseGuards(MerchantGuard)
```
Comment (sandrex-SGV) [inline apps/api/src/products/products.controller.ts]:
```suggestion

```
remove `MerchantGuard` and `@UseGuards(MerchantGuard)`

File: apps/api/src/products/product.guard.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    const request = context.switchToHttp().getRequest();
+
+    const payload = request.user;
+
+    if (payload.usertype === 'merchant') {
+      return true;
+    }
+    throw new UnauthorizedException('Access denied for consumers');
+  }
+}
```
Comment (sandrex-SGV) [inline apps/api/src/products/product.guard.ts]:
```suggestion

```

delete this file

**Review by sandrex-SGV — CHANGES_REQUESTED** [review]
Check suggestions @KezTimple thanks!

**Review by sandrex-SGV — APPROVED** [review]

**Review by TreshAnn — APPROVED** [review]
LGTM!

**Pull request merged**